server/app/routes/resource.js

```diff
@@ -16,23 +18,52 @@ router.param('resourceId', function(req, res, next, id) {
 })
 
 router.get('/', function(req, res, next) {
-	var whereCondition = { status: 'Approved' };
-	if(req.user && req.user.isAdmin) whereCondition= {};
-	Resource.findAll({where: whereCondition})
-	.then(resources => res.send(resources));
+	// var whereCondition = { status: 'approved' };
+	// if(req.user && req.user.isAdmin) whereCondition= {};
+	Resource.findAll()
+	.then(resources => res.send(resources))
+  .catch(next);
 })
 
 router.post('/', function(req,res,next){
-	Resource.create(req.body)
-	.then(function(newResource){
-		res.send(newResource);
-	})
+
+    var topicId = req.body.topicId;
+    delete req.body.topicId;
+    req.body.userId= req.user.dataValues.id;
+    //create resource
+    Resource.create(req.body)
+    .then(function(newResource){
+        return newResource.addTopic(topicId);
+    })
+    .then(function(resource){
+        res.status(201).send(resource);
+    })
+
 });
 
 router.get('/:resourceId', function(req, res, next){
 	res.send(req.resource);
 });
 
+router.get('/:resourceId/flags', function(req, res, next){
+  FlaggedResource.findAll({
+    where: {
+      resourceId: req.resource.id
+    }
+  })
+  .then(flaggedResource => res.status(200).send(flaggedResource));
+
+});
+
+router.delete('/:flagId/flags', function(req, res, next){
+  FlaggedResource.destroy({
```

Comment:
we should change this route to /flags/:flagId.  Otherwise, it may get confused with a /:resourceId route.